cedar-wasm/src/lib.rs

```diff
@@ -1,8 +1,252 @@
+        Ok(principal) => principal,
+        Err(err) => {
+            return json!({
+                "code": 101,
+                "message": format!("[PrincipalErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let action = match EntityUid::from_str(action_str) {
+        Ok(action) => action,
+        Err(err) => {
+            return json!({
+                "code": 102,
+                "message": format!("[ActionErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let resource = match EntityUid::from_str(resource_str) {
+        Ok(resource) => resource,
+        Err(err) => {
+            return json!({
+                "code": 103,
+                "message": format!("[ResourceErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let context = match Context::from_json_str(context_str, None) {
+        Ok(context) => context,
+        Err(err) => {
+            return json!({
+                "code": 104,
+                "message": format!("[ContextErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let policies = match PolicySet::from_str(policies_str) {
+        Ok(policies) => policies,
+        Err(err) => {
+            return json!({
+                "code": 105,
+                "message": format!("[PoliciesErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let entities = match Entities::from_json_str(entities_str, None) {
+        Ok(entities) => entities,
+        Err(err) => {
+            return json!({
+                "code": 106,
+                "message": format!("[EntitiesErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let request = match Request::new(Some(principal), Some(action), Some(resource), context, None) {
+        Ok(request) => request,
+        Err(err) => {
+            return json!({
+                "code": 107,
+                "message": format!("[RequestErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let authorizer = Authorizer::new();
+    let response = authorizer.is_authorized(&request, &policies, &entities);
+
+    // change response to string
+    let decision = response.decision();
+    let diagnostics = response.diagnostics();
+
+    let _reasons = diagnostics.reason();
+    let mut reasons = HashSet::new();
+    for reason in _reasons {
+        reasons.insert(reason.to_string());
+    }
+
+    let _errors = diagnostics.errors();
+    let mut errors = Vec::new();
+    for err in _errors {
+        let error = err.to_string();
+        errors.push(error);
+    }
+
+    json!({
+        "code": 0,
+        "data": {
+            "decision": decision,
+            "reasons": reasons,
+            "errors": errors,
+        }
+    })
+    .to_string()
+}
+
+#[wasm_bindgen(js_name = "validate")]
+pub fn validate(schema_str: &str, policy_str: &str) -> String {
```

Comment:
similar to the above, i was thinking we would just expose the `json_validate` from cedar-policy, along with its input and output structs